packages/redis/src/RedisClient.ts

```diff
@@ -153,4 +160,50 @@ export class RedisClient implements IRequireInitialization, IDisposable {
             span.finish();
         }
     }
+
+    public async xAddObject<T>(
+        context: SpanContext,
+        type: string | IClassType<T>,
+        streamName: string,
+        key: string,
+        body: T,
+        id: string = RedisStreamID.AutoGenerate
+    ): Promise<string> {
+        const db = this.config.db;
+        const span = this.tracer!.startSpan(this.spanOperationName, { childOf: context });
+
+        this.spanLogAndSetTags(span, this.xAddObject.name, db, key, streamName);
+
+        const typeName = this.getTypeName(type);
+
+        const encodedBody = this.encoder.encode({
+            type: typeName,
+            payload: body,
+        });
+
+        const storableValue = Buffer.from(encodedBody).toString("base64");
```

Comment:
Redis strings are binary-safe, so if the underlying library will let us pass Buffers/byte arrays, I would opt for doing that instead of base-64 encoding. 👍

Reply:
I was following the pattern already being used by our `get` wrapper as seen here: https://github.com/dmmulroy/cookie-cutter/blob/redis-output-sink/packages/redis/src/RedisProxy.ts#L94-L97

I could either refactor both (though this could have downstream implications), keep them as base64, or just change xAdd to store the buffer.

Reply:
the base64 encoding is some kind of legacy thing (I don't even remember why we have that). so generally i am all in favor for removing that, but we'd have to make it backwards compatible and ideally the library does things consistently as well (all base64 or none). so maybe adding a config option `base64encode: boolean` (default true) that turns it on/off for get/xadd/set etc might do?

Reply:
Adding a default option is a good plan for backwards compatibility, that was my main concern.  I'll add that now.

Reply:
All good points. If the `redis` package supports it, I like the idea of being able to out-opt of base-64 encoding. It's probably not worthwhile to make the cookie-cutter wrapper infer (e.g. by peeking at the first x bytes) whether the value has been encoded.

Reply:
that's probably overkill and I don't think that would work reliably ... what if your value is a plain string that happens to look like a base64 value, wouldn't be able to decide if this needs to be decoded or not :-)

Reply:
I didn't say it was a _good_ idea :joy:

Reply:
@sklose @chrola I updated the PR to reflect these changes, let me know if I missed or need to change anything

Reply:
JK Missed updating getObject

Reply:
Okay now it should be good